src/QMCHamiltonians/SOECPComponent.h

```diff
@@ -0,0 +1,118 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Ken Esler, kpesler@gmail.com, University of Illinois at Urbana-Champaign
+//                    Jeremy McMinnis, jmcminis@gmail.com, University of Illinois at Urbana-Champaign
+//                    Jeongnim Kim, jeongnim.kim@gmail.com, University of Illinois at Urbana-Champaign
+//                    Jaron T. Krogel, krogeljt@ornl.gov, Oak Ridge National Laboratory
+//                    Mark A. Berrill, berrillma@ornl.gov, Oak Ridge National Laboratory
+//
+// File created by: Jeongnim Kim, jeongnim.kim@gmail.com, University of Illinois at Urbana-Champaign
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_SO_ECPOTENTIAL_COMPONENT_H
+#define QMCPLUSPLUS_SO_ECPOTENTIAL_COMPONENT_H
+#include "QMCHamiltonians/QMCHamiltonianBase.h"
+#include "QMCWaveFunctions/TrialWaveFunction.h"
+#include "Numerics/OneDimGridBase.h"
+#include "Numerics/OneDimGridFunctor.h"
+#include "Numerics/OneDimLinearSpline.h"
+#include "Numerics/OneDimCubicSpline.h"
+#include "Numerics/OhmmsBlas.h"
+
+namespace qmcplusplus
+{
+/** class SOECPComponent
+ **  brief Computes the nonlocal spin-orbit interaction $\Delta V_SO(r) |ljm_j><ljm_j|$.
+ **  details This computes the nonlocal spin-orbit interaction between a single ion species and 
+ **           a given electron.  
+ **           Currently, this class does nothing other than generate and store $\Delta V_SO(r)$
+ **           for different orbital angular momenta.  Implementation coming soon!  
+ **/
+class SOECPComponent : public QMCTraits
+{
+private:
+  typedef OneDimGridBase<RealType> GridType;
+  typedef OneDimCubicSpline<RealType> RadialPotentialType;
+
+  ///Non Local part: angular momentum, potential and grid
+  int lmax;
+  ///the number of non-local channels
+  int nchannel;
+  ///Maximum cutoff the non-local pseudopotential
+  RealType Rmax;
+  ///Angular momentum map
+  aligned_vector<int> angpp_m;
+  ///Non-Local part of the pseudo-potential
+  std::vector<RadialPotentialType*> sopp_m;
+
+  ParticleSet::ParticleGradient_t dG;
+  ParticleSet::ParticleLaplacian_t dL;
+
+public:
+  SOECPComponent(){};
+
+  ///destructor
+  ~SOECPComponent(){};
+
+  SOECPComponent* makeClone(const ParticleSet& qp){};
+
+  ///add a new Spin-Orbit component
+  void add(int l, RadialPotentialType* pp);
+  ///API for accessing the value of an SO radial potential at distance r.  For unit and other testing.
+  RealType test_splined_pot(int l, RealType r);
```

Comment:
This function name should be something like 'getSplinedPot' or 'getSplinedValue'.    Having 'test' in the name indicates the function itself performs a test.    The comment appropriately indicates the function is intended for use in testing.    

The coding standard for the format of function names is to start with a lowercase letter and start subsequent words with uppercase.

Reply:
functions used by tests can be friend functions which live with the tests.

Reply:
I renamed this function in line with Mark's suggestion.  @ye-luo, if you insist on making this a friend function, I will, otherwise renaming is the path of least resistance.